### PR TITLE
Stacked improvements

### PR DESCRIPTION
**Pull request by ferrarafer**
- Added ListenableServiceMixin
- Deprecated ReactiveServiceMixin
- Use ListenableServiceMixin instead of ReactiveServiceMixin
- Change reactiveServices property name on ReactiveViewModels to listenableServices
- Move data models to data_models folder
- Move SkeletonLoader to ui folder
- Rename state_management folder to view_models
- Fix unit test file names to follow conventions

### CODE REVIEW

**Comment by ferrarafer** [comment]
@FilledStacks 
We need to verify how to extends `ReactiveServiceMixin` from `ListenableServiceMixin`.

File: packages/stacked/example/lib/ui/builder_widget_example/builder_widget_example_view.dart [hunk 1]
```diff
@@ -2,25 +2,21 @@ import 'package:flutter/material.dart';
 import 'package:example/ui/home/home_viewmodel.dart';
 import 'package:stacked/stacked.dart';
 
-class BuilderWidgetExampleView extends ViewModelBuilderWidget<HomeViewModel> {
+class BuilderWidgetExampleView extends StackedView<HomeViewModel> {
```
Comment (FilledStacks) [inline packages/stacked/example/lib/ui/builder_widget_example/builder_widget_example_view.dart:5]: Nice 🚀 We should note this down as a breaking change please. 

It needs to go into our migration section.

**Review by FilledStacks — CHANGES_REQUESTED** [review]
Great work! 

We should add a 

### Breaking Changes

Section into the Change log so we can easily identify them and add it to our migration guide when we do our release.

**Review by FilledStacks — APPROVED** [review]
Happy with this. As a note the Breaking Changes title should be an H3 Using `###`

**Pull request merged**